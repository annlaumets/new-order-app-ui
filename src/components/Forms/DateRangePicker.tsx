import { Box, Button, ButtonProps, Icon, Text } from '@chakra-ui/react'
import { forwardRef, useState } from 'react'
import DatePicker from 'react-datepicker'
import { FiCalendar } from 'react-icons/fi'

interface DateRangePickerProps {
  onDateChange: (start: Date, end: Date) => void
  placeholder: string
}

const DateRangePicker: React.FC<DateRangePickerProps> = ({
  placeholder,
  onDateChange,
}) => {
  const [startDate, setStartDate] = useState(null)
  const [endDate, setEndDate] = useState(null)

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const onChange = (dates: any) => {
    const [start, end] = dates
    setStartDate(start)
    setEndDate(end)
    onDateChange(start, end)
  }

  const CustomInput = forwardRef<HTMLButtonElement, ButtonProps>(
    ({ value, onClick }, ref) => {
      return (
        <Button
          data-testid="dateRangePicker"
          value={value}
          ref={ref}
          onClick={onClick}
          fontSize="0.8rem"
          fontWeight="bold"
          variant="outlineGrayRounded"
          px="7"
          leftIcon={
            <Icon
              mr="1"
              fontSize="20"
              fill="transparent"
              stroke="gray.80"
              as={FiCalendar}
            />
          }
        >
          <Text color="gray.60" mr="1" fontSize={['0.6rem', '0.8rem']}>
            {placeholder}
          </Text>
          <Text fontSize={['0.6rem', '0.8rem']}>
            {value || 'mm/dd/yyyy - mm/dd/yyyy'}
          </Text>
        </Button>
      )
    }
  )

  return (
    <Box
      as={DatePicker}
      className="date-range-picker"
      selected={startDate}
      onChange={onChange}
      startDate={startDate}
      endDate={endDate}
      selectsRange
      customInput={<CustomInput />}
      placeholderText="Click to select a date"
      w="auto"
      monthsShown={2}
      dateFormat="MM/dd/yyyy"
      isClearable
    />
  )
}

export default DateRangePicker
