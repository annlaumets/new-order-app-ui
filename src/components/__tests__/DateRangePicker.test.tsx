import { fireEvent, render, waitFor } from '@testing-library/react'

import DateRangePicker from 'components/Forms/DateRangePicker'

describe('CustomDatePicker.tsx component', () => {
  const { container, getByTestId } = render(
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    <DateRangePicker onDateChange={() => {}} placeholder="Vote Launch:" />
  )
  it('Date Range picker', async () => {
    const testDateRangePicker = getByTestId('dateRangePicker')
    fireEvent.click(testDateRangePicker)
    await waitFor(() => {
      return fireEvent.change(testDateRangePicker, {
        target: {
          value: '07/06/2022 - 07/22/2022',
        },
      })
    })
    expect((testDateRangePicker as HTMLInputElement).value).toBe(
      '07/06/2022 - 07/22/2022'
    )
  })
  test('match to snapshot', () => {
    expect(container).toMatchSnapshot()
  })
})
