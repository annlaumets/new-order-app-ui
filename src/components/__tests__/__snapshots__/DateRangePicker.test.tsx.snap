// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDatePicker.tsx component match to snapshot 1`] = `<div />`;
